app/modules.go

```diff
@@ -134,55 +131,28 @@ func (SlashingModule) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
 	return cdc.MustMarshalJSON(genState)
 }
 
-// GenutilModule defines a custom wrapper around the x/genutil module's
-// AppModuleBasic implementation to provide custom genesis state validation.
-type GenutilModule struct {
-	genutil.AppModuleBasic
-}
-
-// ValidateGenesis validates the x/genutil genesis state.
-func (GenutilModule) ValidateGenesis(
-	cdc codec.JSONCodec,
-	encCfg client.TxEncodingConfig,
-	bz json.RawMessage,
-) error {
-	var genState genutiltypes.GenesisState
-	if err := cdc.UnmarshalJSON(bz, &genState); err != nil {
-		return fmt.Errorf("failed to unmarshal %s genesis state: %w", genutiltypes.ModuleName, err)
+func GenTxValidator(msgs []sdk.Msg) error {
+	if n := len(msgs); n != 2 {
+		return fmt.Errorf(
+			"contains invalid number of messages; expected: 2; got: %d", n)
 	}
 
-	txJSONDecoder := encCfg.TxJSONDecoder()
-
-	for i, genTx := range genState.GenTxs {
-		var tx sdk.Tx
-
-		tx, err := txJSONDecoder(genTx)
-		if err != nil {
-			return err
-		}
-
-		msgs := tx.GetMsgs()
-		if n := len(msgs); n != 2 {
-			return fmt.Errorf(
-				"gentx %d contains invalid number of messages; expected: 2; got: %d",
-				i, n,
-			)
-		}
+	if _, ok := msgs[0].(*stakingtypes.MsgCreateValidator); !ok {
+		return fmt.Errorf(
+			"contains invalid message at index 0; expected: %T; got: %T",
+			&stakingtypes.MsgCreateValidator{}, msgs[0])
+	}
 
-		if _, ok := msgs[0].(*stakingtypes.MsgCreateValidator); !ok {
-			return fmt.Errorf(
-				"gentx %d contains invalid message at index 0; expected: %T; got: %T",
-				i, &stakingtypes.MsgCreateValidator{}, msgs[0],
-			)
-		}
+	if _, ok := msgs[1].(*gravitytypes.MsgSetOrchestratorAddress); !ok {
```

Comment:
The e2e peggo debugging + @facundomedica  help makes me think that the [submitted eth address does not match delegate eth address ](https://github.com/umee-network/Gravity-Bridge/blob/81f83566d220edb30b7be55a3a246f597969bed5/module/x/gravity/keeper/msg_server.go#L341) error could be related of us changing the validate or `collect-gentx`

Reply:
I didn't change the logic. Instead we moved the code around. In fact this is required by GB to provide eth address during genesis.

Reply:
@robert-zaremba actually I think you fixed a bug here. In the original implementation, where: 

```golang
	var genState genutiltypes.GenesisState
	if err := cdc.UnmarshalJSON(bz, &genState); err != nil {
		return fmt.Errorf("failed to unmarshal %s genesis state: %w", genutiltypes.ModuleName, err)
	}

	txJSONDecoder := encCfg.TxJSONDecoder()

	for i, genTx := range genState.GenTxs {
		// validation, etc...
```

bz doesn't unmarshal into genState correctly. so, we never actually enter the validation loop since ```len(genState.GenTxs) = 0```

cc: @RafilxTenfen 

this just means we've got to find a way for ignite cli to add MsgSetOrchestratorAddress to gentxs automatically 😉

Reply:
Yes, now we are validating in `collect-gentx`, before we didn't do that. It is related to the gentx rework I described few weeks ago when decided to fork SDK.
But the msg criteria didn't change.